### PR TITLE
filter flash_attn optional imports loading remote code

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

in some remote available models code, we can meet optional import flash_attention module without try-except block. 
Examples:
phi3-vision - https://huggingface.co/microsoft/Phi-3-vision-128k-instruct/blob/main/modeling_phi3_v.py#L52
orion-14b - https://huggingface.co/OrionStarAI/Orion-14B-Chat/blob/main/modeling_orion.py#L36
deepseek-moe - https://huggingface.co/deepseek-ai/deepseek-moe-16b-base/blob/main/modeling_deepseek.py#L54
nanoLLAVA- https://huggingface.co/qnguyen3/nanoLLaVA/blob/main/modeling_llava_qwen2.py#L861

loading such model in environment where flash_attn package is not installed failed with trust_remote_code flag. It may be problematic to install and import this package (e.g. for environment where no cuda and torch installed for cpu only) for some environment. This PR update dependencies search logic for checking dynamic modules loading.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @Rocketknight1 as I think you were working on a related issue

**Comment by andrei-kochin** [comment]
@amyeroberts @Rocketknight1 do you have any good news for us here?

**Comment by itikhono** [comment]
@amyeroberts @Rocketknight1  Hi! any updates on this?

**Comment by amyeroberts** [comment]
Gentle ping @Rocketknight1

File: src/transformers/dynamic_module_utils.py [hunk 1]
```diff
@@ -149,6 +149,8 @@ def get_imports(filename: Union[str, os.PathLike]) -> List[str]:
 
     # filter out try/except block so in custom code we can have try/except imports
     content = re.sub(r"\s*try\s*:\s*.*?\s*except\s*.*?:", "", content, flags=re.MULTILINE | re.DOTALL)
+    # filter out imports under is_flash_attn_2_available block for avoid import issues in cpu only environment
+    content = re.sub(r"if is_flash_attn_*available\(\):\s*(from flash_attn\s*.*\s*)+", "", content, flags=re.MULTILINE)
```
Comment (CuriousPanCake) [inline src/transformers/dynamic_module_utils.py:153]: I believe, we could improve the regex to `r"if is_flash_attn_[0-9]+_available\(\):\s*(from flash_attn\s*.*\s*)+"`
I was trying to use this change locally for OrionStarAI/Orion-14B-Chat and the flash_attn imports were not removed.
Most probably, the similar stands for the other models mentioned in the ticket as the imports string are the same.

Comment (eaidova) [reply src/transformers/dynamic_module_utils.py:153]: sorry, but I lost any motivation to finishing this PR after 3 months o silence... I'm not good in regular expressions, so I have no time to explore all models on hub to find all possible patterns...

I'll apply suggestion from @CuriousPanCake, but do not promise to look anything beyond of that if you do not have specific suggestions.

**Comment by ArthurZucker** [comment]
Ping me if you need help to fix the CI / a review 🤗

**Comment by eaidova** [comment]
@ArthurZucker thank you for review, I fixed code style CI related issue, but I have no idea about flax tests (I do not think that it is some how related to my changes)

**Review by Rocketknight1 — APPROVED** [review]
Hey! The fault for the extremely slow review is mine, and I'm sorry, especially because this issue affected other users besides you, and I really should have gotten it merged before now!

The tl;dr is that I think it's ready for merge as-is, because the only relevant test in our codebase is `is_flash_attn_2_available`. However, I've suggested a change that should catch anything that looks like `is_flash_attn[...]available`, which should cover us if any custom code is calling something similar, but without causing unwanted side effects.

Let me know if you're happy with the change, and then we'll try to merge this ASAP - you definitely don't need to trawl through the code after having to wait all this time. Thank you for the PR!

**Comment by Rocketknight1** [comment]
@eaidova thanks for committing the suggestion, merging now!

**Pull request merged**